File: ch01/1.16/main.hs
type Pair a b = (a, b)

invert :: [Pair a b] -> [Pair b a]
invert = map (\p -> (snd p, fst p))

main :: IO ()
main =
  print ("invert([('a', 1), ('a', 2), (1, 'b'), (2, 'b')]): " ++ show (invert [('a', '1'), ('a', '2'), ('1', 'b'), ('2', 'b')]))

File: ch01/1.16/main.go
package main

import "fmt"

type Pair struct {
	a, b interface{}
}

func invert(xs []Pair) []Pair {
	ys := make([]Pair, len(xs))
	for i, p := range xs {
		ys[i] = Pair{a: p.b, b: p.a}
	}

	return ys
}

func main() {
	xs := []Pair{{"a", 1}, {"a", 2}, {1, "b"}, {2, "b"}}

	fmt.Println(invert(xs))
}
